scripts/create-status-page

```diff
@@ -0,0 +1,284 @@
+import sys
+import urllib.parse
+from typing import Any, Dict, List, Optional
+
+import dateutil.parser
+import markdown
+import requests
+import yaml
+from markdown.extensions.tables import TableExtension
+
+from markdown_table import render
+
+
+class Status:
+    headers: List[str]
+    data: List[List[str]]
+    _row: Optional[List[str]] = None
+
+    def __init__(self, status=None):
+        self.headers = []
+        self.data = []
+
+        if status is None:
+            self.headers = ["Version", "CI status"]
+            return
+
+        md = markdown.Markdown(extensions=[TableExtension()])
+
+        elem = md.parser.parseDocument([s for s in status.split("\n") if s != "" and s[0] != "#" and s[0] != "["])  # type: ignore
+        self._pe(elem.getroot())
+
+        self.data = [r for r in self.data if len([c for c in r if c != None]) > 0]
+        self.headers.append("CI status")
+        for row in self.data:
+            row.append("")
+
+    def _pe(self, elem):
+        if elem.tag == "th":
+            self.headers.append(elem.text)
+        if elem.tag == "tr":
+            self._row = []
+            self.data.append(self._row)
+        if elem.tag == "td":
+            self._row.append(elem.text)  # type: ignore
+        for e in elem.getchildren():
+            self._pe(e)
+
+    def index(self, ver, link):
+        for ind, row in enumerate(self.data):
+            if row[0] == ver:
+                row[0] = link
+                return ind
+        row = ["" for _ in self.headers]
+        row[0] = link
+        self.data.append(row)
+        return len(self.data) - 1
+
+    def render(self):
+        return render(self.headers, self.data)
+
+
+AUTH_HEADER = ""
+
+
+def main():
+    global AUTH_HEADER
+    AUTH_HEADER = "Bearer {}".format(
+        os.environ["GITHUB_TOKEN"]
+        if "GITHUB_TOKEN" in os.environ
+        else subprocess.check_output(["gopass", "gs/ci/github/token/gopass"]).strip().decode()
+    )
+    with open(os.path.join(os.path.dirname(__file__), "status.yaml")) as f:
+        config = yaml.load(f, Loader=yaml.SafeLoader)
+        for elem in config:
+            project(elem)
+
+    print()
+    print("Created with `scripts/create-status-page`")
+
+
+def project(elem):
+    repo = elem["repo"]
+    all_workflows_responce = requests.get(
+        "https://api.github.com/repos/{repo}/actions/workflows".format(repo=repo),
+        headers={"Accept": "application/vnd.github.v3+json", "Authorization": AUTH_HEADER},
+    )
+    if all_workflows_responce.status_code == 404:
+        all_workflows = []
+    else:
+        all_workflows_responce.raise_for_status()
+        all_workflows = all_workflows_responce.json()["workflows"]
+    all_workflows_name = {w["name"] for w in all_workflows if w["name"] != ""}
+    all_workflows_by_name: Dict[str, Any] = {}
+    for w in all_workflows:
+        all_workflows_by_name.setdefault(w["name"], [])
+        all_workflows_by_name[w["name"]].append(w)
```

Comment:
This could use `itertools.groupby` ex:
```
all_workflows_by_name = {
    name: workflows
    for name, worflows in itertools.groupby(all_workflows, lambda w: w["name"])
}
```
or maybe:
```
all_workflows_by_name = dict(itertools.groupby(all_workflows, lambda w: w["name"]))
```